### ISSUE TITLE
Guest fail to start on minikube with driver=podman

### ISSUE THREAD

**Issue by bkircher** [kubevirt/kubevirt#5983]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What happened**:

tl;dr: Using minikube with *--driver=podman* fails to start guests. Installation of KubeVirt works fine, though.

minikube with podman and CRI-O fails to start guests when following [lab 1](https://kubevirt.io/labs/kubernetes/lab1.html). minikube with kvm2 and nested virtualization works fine.

**What you expected to happen**:

Not sure if minikube with podman driver is supported? minikube with --driver=kvm2 and nested virt works fine. Maybe this just needs more clarification in the minikube quick start guide over at [kubevirt.io/quickstart_minikube](https://kubevirt.io/quickstart_minikube/)?

**How to reproduce it (as minimally and precisely as possible)**:

Make sure to have podman and CRI-O runtime installed (and also kubectl). On a Fedora:

```
$ dnf module list cri-o
$ sudo dnf module enable cri-o:1.20
$ sudo dnf install cri-o podman kubernetes-client
```

First start that minikube with podman driver
```
$ minikube start \
  --driver=podman \
  --container-runtime=cri-o \
  --extra-config=kubelet.cgroup-driver=systemd \
  --feature-gates="LocalStorageCapacityIsolation=false"
```

Then just install KubeVirt using the operator and follow [lab 1](https://kubevirt.io/labs/kubernetes/lab1.html).

Operator and resource definitions:
```
$ echo $KUBEVIRT_VERSION
v0.42.1

$ kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/$KUBEVIRT_VERSION/kubevirt-operator.yaml
$ kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/$KUBEVIRT_VERSION/kubevirt-cr.yaml
```

When starting the testvm:
```
$ virtctl start testvm                                       
VM testvm was scheduled to start


$ kubectl get vms -w                                                                                               
NAME     AGE   STATUS    VOLUME
testvm   14s   Stopped   
testvm   23s   Stopped   
testvm   23s   Starting   

```

Will stay in starting state. Logs below.

First error in handler logs:
```
{"component":"virt-handler","kind":"","level":"error","msg":"Synchronizing the VirtualMachineInstance failed.","name":"testvm","namespace":"default","pos":"vm.go:1654","reason":"server error. command SyncVMI failed: \"LibvirtError(Code=1, Domain=10, Message='internal error: qemu unexpectedly closed the monitor: 2021-07-07T08:57:37.549018Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.5,addr=0x0,drive=libvirt-1-format,id=ua-cloudinitdisk,write-cache=on,werror=stop,rerror=stop: Cannot get 'write' permission without 'resize': Image size is not a multiple of request alignment')\"","timestamp":"2021-07-07T08:57:37.754784Z","uid":"a3b31b16-0c7c-4144-be1d-b5a40b3bccab"}
```
launcher logs contain multiple errors, including above, starting with /dev/mem not being available
```
{"component":"virt-launcher","level":"error","msg":"internal error: Child process (dmidecode -q -t 0,1,2,3,4,11,17) unexpected exit status 1: /dev/mem: No such file or directory","pos":"virCommandWait:2767","subcomponent":"libvirt","thread":"46","timestamp":"2021-07-07T08:59:10.400000Z"}
{"component":"virt-launcher","level":"info","msg":"Connected to libvirt daemon","pos":"libvirt.go:508","timestamp":"2021-07-07T08:59:10.893832Z"}
{"component":"virt-launcher","level":"info","msg":"Registered libvirt event notify callback","pos":"client.go:475","timestamp":"2021-07-07T08:59:10.901363Z"}
{"component":"virt-launcher","level":"info","msg":"Marked as ready","pos":"virt-launcher.go:73","timestamp":"2021-07-07T08:59:10.902463Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Executing PreStartHook on VMI pod environment","name":"testvm","namespace":"default","pos":"manager.go:423","timestamp":"2021-07-07T08:59:12.183999Z","uid":"df27c877-0e83-4c71-9738-e0ff2d4a38e2"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Starting PreCloudInitIso hook","name":"testvm","namespace":"default","pos":"manager.go:432","timestamp":"2021-07-07T08:59:12.184038Z","uid":"df27c877-0e83-4c71-9738-e0ff2d4a38e2"}
{"component":"virt-launcher","level":"info","msg":"Found nameservers in /etc/resolv.conf: \n`\u0000\n","pos":"network.go:273","timestamp":"2021-07-07T08:59:12.184521Z"}
{"component":"virt-launcher","level":"info","msg":"Found search domains in /etc/resolv.conf: default.svc.cluster.local svc.cluster.local cluster.local dns.podman","pos":"network.go:274","timestamp":"2021-07-07T08:59:12.184545Z"}
{"component":"virt-launcher","level":"info","msg":"Starting SingleClientDHCPServer","pos":"dhcp.go:64","timestamp":"2021-07-07T08:59:12.184581Z"}
{"component":"virt-launcher","level":"info","msg":"Driver cache mode for /var/run/kubevirt-ephemeral-disks/disk-data/containerdisk/disk.qcow2 set to none","pos":"converter.go:396","timestamp":"2021-07-07T08:59:12.211180Z"}
{"component":"virt-launcher","level":"error","msg":"Error when closing file","pos":"os_helper.go:36","reason":"invalid argument","timestamp":"2021-07-07T08:59:12.214290Z"}
{"component":"virt-launcher","level":"info","msg":"Driver cache mode for /var/run/kubevirt-ephemeral-disks/cloud-init-data/default/testvm/noCloud.iso set to none","pos":"converter.go:396","timestamp":"2021-07-07T08:59:12.214331Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Domain XML generated. Base64 dump PGRvbWFpbiB0eXBlPSJrdm0iIHhtbG5zOnFlbXU9Imh0dHA6Ly9saWJ2aXJ0Lm9yZy9zY2hlbWFzL2RvbWFpbi9xZW11LzEuMCI+Cgk8bmFtZT5kZWZhdWx0X3Rlc3R2bTwvbmFtZT4KCTxtZW1vcnkgdW5pdD0iYiI+NjQwMDAwMDA8L21lbW9yeT4KCTxvcz4KCQk8dHlwZSBhcmNoPSJ4ODZfNjQiIG1hY2hpbmU9InEzNSI+aHZtPC90eXBlPgoJCTxzbWJpb3MgbW9kZT0ic3lzaW5mbyI+PC9zbWJpb3M+Cgk8L29zPgoJPHN5c2luZm8gdHlwZT0ic21iaW9zIj4KCQk8c3lzdGVtPgoJCQk8ZW50cnkgbmFtZT0idXVpZCI+NWE5ZmMxODEtOTU3ZS01YzMyLTllNWEtMmRlNWU5NjczNTMxPC9lbnRyeT4KCQkJPGVudHJ5IG5hbWU9Im1hbnVmYWN0dXJlciI+S3ViZVZpcnQ8L2VudHJ5PgoJCQk8ZW50cnkgbmFtZT0iZmFtaWx5Ij5LdWJlVmlydDwvZW50cnk+CgkJCTxlbnRyeSBuYW1lPSJwcm9kdWN0Ij5Ob25lPC9lbnRyeT4KCQkJPGVudHJ5IG5hbWU9InNrdSI+PC9lbnRyeT4KCQkJPGVudHJ5IG5hbWU9InZlcnNpb24iPjwvZW50cnk+CgkJPC9zeXN0ZW0+CgkJPGJpb3M+PC9iaW9zPgoJCTxiYXNlQm9hcmQ+PC9iYXNlQm9hcmQ+CgkJPGNoYXNzaXM+PC9jaGFzc2lzPgoJPC9zeXNpbmZvPgoJPGRldmljZXM+CgkJPGludGVyZmFjZSB0eXBlPSJldGhlcm5ldCI+CgkJCTxzb3VyY2U+PC9zb3VyY2U+CgkJCTx0YXJnZXQgZGV2PSJ0YXAwIiBtYW5hZ2VkPSJubyI+PC90YXJnZXQ+CgkJCTxtb2RlbCB0eXBlPSJ2aXJ0aW8tbm9uLXRyYW5zaXRpb25hbCI+PC9tb2RlbD4KCQkJPG1hYyBhZGRyZXNzPSI4NjpkZDplYzpjOTowZDozMiI+PC9tYWM+CgkJCTxtdHUgc2l6ZT0iMTUwMCI+PC9tdHU+CgkJCTxhbGlhcyBuYW1lPSJ1YS1kZWZhdWx0Ij48L2FsaWFzPgoJCQk8cm9tIGVuYWJsZWQ9Im5vIj48L3JvbT4KCQk8L2ludGVyZmFjZT4KCQk8Y2hhbm5lbCB0eXBlPSJ1bml4Ij4KCQkJPHRhcmdldCBuYW1lPSJvcmcucWVtdS5ndWVzdF9hZ2VudC4wIiB0eXBlPSJ2aXJ0aW8iPjwvdGFyZ2V0PgoJCTwvY2hhbm5lbD4KCQk8Y29udHJvbGxlciB0eXBlPSJ1c2IiIGluZGV4PSIwIiBtb2RlbD0ibm9uZSI+PC9jb250cm9sbGVyPgoJCTxjb250cm9sbGVyIHR5cGU9InNjc2kiIGluZGV4PSIwIiBtb2RlbD0idmlydGlvLW5vbi10cmFuc2l0aW9uYWwiPjwvY29udHJvbGxlcj4KCQk8Y29udHJvbGxlciB0eXBlPSJ2aXJ0aW8tc2VyaWFsIiBpbmRleD0iMCIgbW9kZWw9InZpcnRpby1ub24tdHJhbnNpdGlvbmFsIj48L2NvbnRyb2xsZXI+CgkJPHZpZGVvPgoJCQk8bW9kZWwgdHlwZT0idmdhIiBoZWFkcz0iMSIgdnJhbT0iMTYzODQiPjwvbW9kZWw+CgkJPC92aWRlbz4KCQk8Z3JhcGhpY3MgdHlwZT0idm5jIj4KCQkJPGxpc3RlbiB0eXBlPSJzb2NrZXQiIHNvY2tldD0iL3Zhci9ydW4va3ViZXZpcnQtcHJpdmF0ZS9kZjI3Yzg3Ny0wZTgzLTRjNzEtOTczOC1lMGZmMmQ0YTM4ZTIvdmlydC12bmMiPjwvbGlzdGVuPgoJCTwvZ3JhcGhpY3M+CgkJPG1lbWJhbGxvb24gbW9kZWw9InZpcnRpby1ub24tdHJhbnNpdGlvbmFsIj4KCQkJPHN0YXRzIHBlcmlvZD0iMTAiPjwvc3RhdHM+CgkJPC9tZW1iYWxsb29uPgoJCTxkaXNrIGRldmljZT0iZGlzayIgdHlwZT0iZmlsZSIgbW9kZWw9InZpcnRpby1ub24tdHJhbnNpdGlvbmFsIj4KCQkJPHNvdXJjZSBmaWxlPSIvdmFyL3J1bi9rdWJldmlydC1lcGhlbWVyYWwtZGlza3MvZGlzay1kYXRhL2NvbnRhaW5lcmRpc2svZGlzay5xY293MiI+PC9zb3VyY2U+CgkJCTx0YXJnZXQgYnVzPSJ2aXJ0aW8iIGRldj0idmRhIj48L3RhcmdldD4KCQkJPGRyaXZlciBjYWNoZT0ibm9uZSIgZXJyb3JfcG9saWN5PSJzdG9wIiBuYW1lPSJxZW11IiB0eXBlPSJxY293MiIgZGlzY2FyZD0idW5tYXAiPjwvZHJpdmVyPgoJCQk8YWxpYXMgbmFtZT0idWEtY29udGFpbmVyZGlzayI+PC9hbGlhcz4KCQkJPGJhY2tpbmdTdG9yZSB0eXBlPSJmaWxlIj4KCQkJCTxmb3JtYXQgdHlwZT0icWNvdzIiPjwvZm9ybWF0PgoJCQkJPHNvdXJjZSBmaWxlPSIvdmFyL3J1bi9rdWJldmlydC9jb250YWluZXItZGlza3MvZGlza18wLmltZyI+PC9zb3VyY2U+CgkJCTwvYmFja2luZ1N0b3JlPgoJCTwvZGlzaz4KCQk8ZGlzayBkZXZpY2U9ImRpc2siIHR5cGU9ImZpbGUiIG1vZGVsPSJ2aXJ0aW8tbm9uLXRyYW5zaXRpb25hbCI+CgkJCTxzb3VyY2UgZmlsZT0iL3Zhci9ydW4va3ViZXZpcnQtZXBoZW1lcmFsLWRpc2tzL2Nsb3VkLWluaXQtZGF0YS9kZWZhdWx0L3Rlc3R2bS9ub0Nsb3VkLmlzbyI+PC9zb3VyY2U+CgkJCTx0YXJnZXQgYnVzPSJ2aXJ0aW8iIGRldj0idmRiIj48L3RhcmdldD4KCQkJPGRyaXZlciBjYWNoZT0ibm9uZSIgZXJyb3JfcG9saWN5PSJzdG9wIiBuYW1lPSJxZW11IiB0eXBlPSJyYXciIGRpc2NhcmQ9InVubWFwIj48L2RyaXZlcj4KCQkJPGFsaWFzIG5hbWU9InVhLWNsb3VkaW5pdGRpc2siPjwvYWxpYXM+CgkJPC9kaXNrPgoJCTxzZXJpYWwgdHlwZT0idW5peCI+CgkJCTx0YXJnZXQgcG9ydD0iMCI+PC90YXJnZXQ+CgkJCTxzb3VyY2UgbW9kZT0iYmluZCIgcGF0aD0iL3Zhci9ydW4va3ViZXZpcnQtcHJpdmF0ZS9kZjI3Yzg3Ny0wZTgzLTRjNzEtOTczOC1lMGZmMmQ0YTM4ZTIvdmlydC1zZXJpYWwwIj48L3NvdXJjZT4KCQk8L3NlcmlhbD4KCQk8Y29uc29sZSB0eXBlPSJwdHkiPgoJCQk8dGFyZ2V0IHR5cGU9InNlcmlhbCIgcG9ydD0iMCI+PC90YXJnZXQ+CgkJPC9jb25zb2xlPgoJPC9kZXZpY2VzPgoJPG1ldGFkYXRhPgoJCTxrdWJldmlydCB4bWxucz0iaHR0cDovL2t1YmV2aXJ0LmlvIj4KCQkJPHVpZD5kZjI3Yzg3Ny0wZTgzLTRjNzEtOTczOC1lMGZmMmQ0YTM4ZTI8L3VpZD4KCQkJPGdyYWNlcGVyaW9kPgoJCQkJPGRlbGV0aW9uR3JhY2VQZXJpb2RTZWNvbmRzPjMwPC9kZWxldGlvbkdyYWNlUGVyaW9kU2Vjb25kcz4KCQkJPC9ncmFjZXBlcmlvZD4KCQk8L2t1YmV2aXJ0PgoJPC9tZXRhZGF0YT4KCTxmZWF0dXJlcz4KCQk8YWNwaT48L2FjcGk+Cgk8L2ZlYXR1cmVzPgoJPGNwdSBtb2RlPSJob3N0LW1vZGVsIj4KCQk8dG9wb2xvZ3kgc29ja2V0cz0iMSIgY29yZXM9IjEiIHRocmVhZHM9IjEiPjwvdG9wb2xvZ3k+Cgk8L2NwdT4KCTx2Y3B1IHBsYWNlbWVudD0ic3RhdGljIj4xPC92Y3B1PgoJPGlvdGhyZWFkcz4xPC9pb3RocmVhZHM+CjwvZG9tYWluPg==","name":"testvm","namespace":"default","pos":"libvirt_helper.go:103","timestamp":"2021-07-07T08:59:12.217599Z","uid":"df27c877-0e83-4c71-9738-e0ff2d4a38e2"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Domain defined.","name":"testvm","namespace":"default","pos":"manager.go:771","timestamp":"2021-07-07T08:59:12.822294Z","uid":"df27c877-0e83-4c71-9738-e0ff2d4a38e2"}
{"component":"virt-launcher","level":"info","msg":"DomainLifecycle event 0 with reason 0 received","pos":"client.go:401","timestamp":"2021-07-07T08:59:12.822414Z"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Shutoff(5):Unknown(0)","pos":"client.go:262","timestamp":"2021-07-07T08:59:12.823620Z"}
{"component":"virt-launcher","level":"info","msg":"Successfully connected to domain notify socket at /var/run/kubevirt/domain-notify-pipe.sock","pos":"client.go:151","timestamp":"2021-07-07T08:59:12.824932Z"}
{"component":"virt-launcher","level":"info","msg":"Domain name event: default_testvm","pos":"client.go:377","timestamp":"2021-07-07T08:59:12.827423Z"}
{"component":"virt-launcher","level":"info","msg":"Monitoring loop: rate 1s start timeout 5m0s","pos":"monitor.go:166","timestamp":"2021-07-07T08:59:12.828036Z"}
{"component":"virt-launcher","level":"info","msg":"generated nocloud iso file /var/run/kubevirt-ephemeral-disks/cloud-init-data/default/testvm/noCloud.iso","pos":"cloud-init.go:569","timestamp":"2021-07-07T08:59:12.828124Z"}
{"component":"virt-launcher","level":"error","msg":"At least one cgroup controller is required: No such device or address","pos":"virCgroupDetectControllers:454","subcomponent":"libvirt","thread":"35","timestamp":"2021-07-07T08:59:12.837000Z"}
{"component":"virt-launcher","level":"error","msg":"Unable to read from monitor: Connection reset by peer","pos":"qemuMonitorIORead:494","subcomponent":"libvirt","thread":"87","timestamp":"2021-07-07T08:59:12.859000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: qemu unexpectedly closed the monitor: 2021-07-07T08:59:12.859150Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.5,addr=0x0,drive=libvirt-1-format,id=ua-cloudinitdisk,write-cache=on,werror=stop,rerror=stop: Cannot get 'write' permission without 'resize': Image size is not a multiple of request alignment","pos":"qemuProcessReportLogError:2156","subcomponent":"libvirt","thread":"87","timestamp":"2021-07-07T08:59:12.859000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: process exited while connecting to monitor: 2021-07-07T08:59:12.859150Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.5,addr=0x0,drive=libvirt-1-format,id=ua-cloudinitdisk,write-cache=on,werror=stop,rerror=stop: Cannot get 'write' permission without 'resize': Image size is not a multiple of request alignment","pos":"qemuProcessReportLogError:2156","subcomponent":"libvirt","thread":"35","timestamp":"2021-07-07T08:59:13.060000Z"}
{"component":"virt-launcher","kind":"","level":"error","msg":"Failed to start VirtualMachineInstance with flags 0.","name":"testvm","namespace":"default","pos":"manager.go:796","reason":"virError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: 2021-07-07T08:59:12.859150Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.5,addr=0x0,drive=libvirt-1-format,id=ua-cloudinitdisk,write-cache=on,werror=stop,rerror=stop: Cannot get 'write' permission without 'resize': Image size is not a multiple of request alignment')","timestamp":"2021-07-07T08:59:13.065187Z","uid":"df27c877-0e83-4c71-9738-e0ff2d4a38e2"}

```

Complete logs attached:
[virt-handler-logs.txt](https://github.com/kubevirt/kubevirt/files/6775880/virt-handler-logs.txt)
[virt-launcher-testvm-logs.txt](https://github.com/kubevirt/kubevirt/files/6775881/virt-launcher-testvm-logs.txt)

**Anything else we need to know?**:

**Environment**: Fedora 34, minikube v1.21.0, podman, CRI-O runtime
- KubeVirt version (use `virtctl version`):
```
Client Version: version.Info{GitVersion:"v0.42.1", GitCommit:"3e4ea2d9b8d4405dc307e885bc49f10c8ce5372c", GitTreeState:"clean", BuildDate:"2021-06-10T01:42:11Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"v0.42.1-dirty", GitCommit:"3e4ea2d9b8d4405dc307e885bc49f10c8ce5372c", GitTreeState:"dirty", BuildDate:"2021-06-10T02:06:01Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"archive", BuildDate:"2021-03-30T00:00:00Z", GoVersion:"go1.16", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7", GitCommit:"132a687512d7fb058d0f5890f07d4121b3f0a2e2", GitTreeState:"clean", BuildDate:"2021-05-12T12:32:49Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
```

- VM or VMI specifications:
```
https://kubevirt.io/labs/manifests/vm.yaml
```
- Cloud provider or hardware configuration: bare-metal, x86_64, kvm_intel
- OS (e.g. from /etc/os-release): Fedora
- Kernel (e.g. `uname -a`):
```
Linux localhost 5.12.13-300.fc34.x86_64 #1 SMP Wed Jun 23 16:18:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools: KubeVirt operator
- Others:
```
$ getenforce          
Permissive
```

**Comment by rmohr** [comment]
Oh I see 

```
{"component":"virt-launcher","level":"error","msg":"internal error: qemu unexpectedly closed the monitor: 2021-07-07T08:59:12.859150Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.5,addr=0x0,drive=libvirt-1-format,id=ua-cloudinitdisk,write-cache=on,werror=stop,rerror=stop: Cannot get 'write' permission without 'resize': Image size is not a multiple of request alignment","pos":"qemuProcessReportLogError:2156","subcomponent":"libvirt","thread":"87","timestamp":"2021-07-07T08:59:12.859000Z"}
```

I think that is resolved with https://github.com/kubevirt/kubevirt/pull/5806.

I think this is a duplicate of https://github.com/kubevirt/kubevirt/issues/5785 and it should be part of 0.43.

**Comment by bkircher** [comment]
Yep! You're right. Works with latest release candidate 

```
~ % virtctl start testvm
VM testvm was scheduled to start
~ % kubectl get vms -w  
NAME     AGE   STATUS     VOLUME
testvm   29s   Starting   
testvm   38s   Running    
testvm   38s   Running    
^C%                                                                                                                                 ~ % echo $KUBEVIRT_VERSION                                                             
v0.43.0-rc.0
```
Thanks!
